### PR TITLE
Re-added the account info paper on creating a new account.

### LINKED ISSUE #22200: Unable to obtain PIN for new accounts
**Issue Description**:

When creating an account at the accounts uplink terminal, no paper gets printed like it did in the past (iirc). There is no button that I could find in the accounts uplink terminal, the nanobank app or the atm that shows you the pin. To change the pin you need to know the existing one.

**What did you expect to happen**: 

There should be some way to know the pin of a newly created account, for example via a paper printout at account creation.

**What happened instead**:

Nothing

**Why is this bad/What are the consequences**:

Setting access to pin only for an account where you don't know the pin locks you out of it.

**Steps to reproduce the problem**:

Create new account at accounts uplink terminal.

**When did the problem start happening**:

Didn't play for a while, so I can't give you a date.

**Extra information**:

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
We used to print out a wrapped sheet of paper with account details when creating a new account, but it got lost in the economy rework. Re-added.
Fixed #22200.

## Why It's Good For The Game
You should have PIN access to a newly-created account.

## Images of changes
image
image

## Testing
Became HoP. Made definitely-not-illegal new account. Got nice printout. Was able to update PIN with the provided one.

## Changelog
:cl:
add: Re-added the account info printout on creating a new account at the Accounts Uplink Terminal.
/:cl:

### CODE REVIEW

**Comment by Danchi299** [comment]
now we just need for captain level ID to check the password of department accounts so acting heads can access it

File: code/modules/economy/economy_machinery/account_terminal.dm [hunk 1]
```diff
@@ -136,9 +136,41 @@
...
 			if(!account_name)
 				return
-			GLOB.station_money_database.create_account(account_name, 0, ACCOUNT_SECURITY_ID, name, supress_log = FALSE)
+			var/datum/money_account/account = GLOB.station_money_database.create_account(account_name, 0, ACCOUNT_SECURITY_ID, name, supress_log = FALSE)
+			print_new_account_info(account)
 			current_page = AUT_ACCLST
 
+/obj/machinery/computer/account_database/proc/print_new_account_info(datum/money_account/account)
+	//create a sealed package containing the account details
+	var/obj/item/smallDelivery/P = new /obj/item/smallDelivery(loc)
```
Comment (warriorstar-orion) [inline code/modules/economy/economy_machinery/account_terminal.dm:145]: can you use more descriptive variable names than `P` and `R`?

File: code/modules/economy/economy_machinery/account_terminal.dm [hunk 2]
```diff
@@ -136,9 +136,41 @@
...
+		<i>Account holder:</i> [account.account_name]<br>
+		<i>Account number:</i> [account.account_number]<br>
+		<i>Account pin:</i> [account.account_pin]<br>
+		<i>Starting balance:</i> $[account.credit_balance]<br>
+		<i>Date and time:</i> [station_time_timestamp()], [GLOB.current_date_string]<br><br>
+		<i>Authorised NT officer overseeing creation:</i> [overseer]<br>"}
+
+	//stamp the paper
+	var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+	stampoverlay.icon_state = "paper_stamp-cent"
```
Comment (warriorstar-orion) [inline code/modules/economy/economy_machinery/account_terminal.dm:166]: I don't see the actual stamp overlay on the screenshot.

File: code/modules/economy/economy_machinery/account_terminal.dm [hunk 3]
```diff
@@ -136,9 +136,41 @@
...
+	R.name = "Account information: [account.account_name]"
+
+	var/overseer = "Unknown"
+	var/datum/ui_login/L = ui_login_get()
+	if(L.id)
+		overseer = L.id.registered_name
+	R.info = {"<b>Account details (confidential)</b><br><hr><br>
+		<i>Account holder:</i> [account.account_name]<br>
+		<i>Account number:</i> [account.account_number]<br>
+		<i>Account pin:</i> [account.account_pin]<br>
```
Comment (warriorstar-orion) [inline code/modules/economy/economy_machinery/account_terminal.dm:159]:
```suggestion
		<i>Account PIN:</i> [account.account_pin]<br>
```

File: code/modules/economy/economy_machinery/account_terminal.dm [hunk 4]
```diff
@@ -136,9 +136,41 @@
...
+		<i>Starting balance:</i> $[account.credit_balance]<br>
+		<i>Date and time:</i> [station_time_timestamp()], [GLOB.current_date_string]<br><br>
```
Comment (warriorstar-orion) [inline code/modules/economy/economy_machinery/account_terminal.dm:161]: I would swap the order of the time and date here, and also just call the field something like "Creation date"

**Review by warriorstar-orion — CHANGES_REQUESTED** [review]

File: code/modules/economy/economy_machinery/account_terminal.dm [hunk 5]
```diff
@@ -136,9 +136,41 @@
...
+			var/datum/money_account/account = GLOB.station_money_database.create_account(account_name, 0, ACCOUNT_SECURITY_ID, name, supress_log = FALSE)
+			print_new_account_info(account)
 			current_page = AUT_ACCLST
 
+/obj/machinery/computer/account_database/proc/print_new_account_info(datum/money_account/account)
+	//create a sealed package containing the account details
+	var/obj/item/smallDelivery/P = new /obj/item/smallDelivery(loc)
+
+	var/obj/item/paper/R = new /obj/item/paper(P)
+	playsound(loc, 'sound/goonstation/machines/printer_thermal.ogg', 50, 1)
```
Comment (Burzah) [inline code/modules/economy/economy_machinery/account_terminal.dm:148]:
```suggestion
	playsound(loc, 'sound/goonstation/machines/printer_thermal.ogg', 50, TRUE)
```

File: code/modules/economy/economy_machinery/account_terminal.dm [hunk 6]
```diff
@@ -136,9 +136,41 @@
...
+		<i>Authorised NT officer overseeing creation:</i> [overseer]<br>"}
+
+	//stamp the paper
+	var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+	stampoverlay.icon_state = "paper_stamp-cent"
+	if(!R.stamped)
+		R.stamped = new
+	R.stamped += /obj/item/stamp
+	R.stamp_overlays += stampoverlay
+	R.stamps += "<HR><i>This paper has been stamped by the Accounts Database.</i>"
```
Comment (Burzah) [inline code/modules/economy/economy_machinery/account_terminal.dm:171]:
```suggestion
	R.stamps += "<hr><i>This paper has been stamped by the Accounts Database.</i>"
```

**Review by Burzah — CHANGES_REQUESTED** [review]

File: code/modules/economy/economy_machinery/account_terminal.dm (hunk 2 above)
Comment (FunnyMan3595) [reply code/modules/economy/economy_machinery/account_terminal.dm:166]: That confused me, too. Updated the comment to reflect that this is the stamp on the *icon*, not the full printout.

**Review by S34NW — APPROVED** [review]

**Pull request merged**